src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs

```diff
@@ -129,26 +134,47 @@ private static int KMPSearch(byte[] pattern, byte[] bytes)
                     }
                 }
             }
+
+            accessor.SafeMemoryMappedViewHandle.ReleasePointer();
```

Comment:
Why not put this in a finally?

Reply:
there is not try-catch why `finally`? And this method is very procedure (due to perf), `finally` won't have any extra meaning

Reply:
You should put ReleasePointer in a finally regardless. Read from memory mapped file could throw, or there could be an asynchronous exception like thread abort. This is a case where there's a clear pattern, and there's no harm to following the pattern always. Reasons why you think finally isn't required today can be invalidated as the code evolves, and diff won't draw attention to it in subsequent reviews.

Reply:
You should also check if ptr is null before calling Release. We want to be as good a citizen as [`ReadByte`](https://source.dot.net/#System.Private.CoreLib/shared/System/IO/UnmanagedMemoryAccessor.cs,afe8e26be70dbb5e,references), but just hoist the Acquire/Release to cover multiple bytes.

Reply:
And added null check before ReleasePointer